src/assets/custom/js/websiteHeader.test.js

```diff
@@ -37,27 +29,84 @@ describe("Website Header", () => {
-        expect(header.classList).toContain("website-header");
-        expect(header.classList).toContain("website-header--revealed");
-        expect(header.classList.length).toBe(2);
       });
 
-      describe("When we scroll all the way up to the top of the page", () => {
+      describe("When we then scroll up the page by just a couple of pixel", () => {
         beforeEach(() => {
-          simulateScrollingToY(0);
+          simulateScrollingToY(1000 - 2);
         });
 
-        it("we let the header return to it's natural position", () => {
+        it("we don't immediately reveal the header", () => {
           expect(header.classList).toContain("website-header");
+          expect(header.classList).not.toContain("website-header--revealed");
           expect(header.classList.length).toBe(1);
         });
       });
+
+      describe("When we then scroll up the page by at least three pixels", () => {
+        beforeEach(() => {
+          simulateScrollingToY(1000 - 3);
+        });
+
+        it("we reveal the header positioned at the top of the viewport", () => {
+          expect(header.classList).toContain("website-header");
+          expect(header.classList).toContain("website-header--revealed");
+          expect(header.classList.length).toBe(2);
+        });
+
+        describe("When we then scroll close to the top of the page", () => {
+          beforeEach(() => {
+            simulateScrollingToY(1);
+          });
+
+          it("we let the header return to it's natural position", () => {
+            expect(header.classList).toContain("website-header");
+            expect(header.classList.length).toBe(1);
+          });
+        });
+
+        describe("When we then scroll down the page by a couple of pixels", () => {
+          beforeEach(() => {
+            simulateScrollingToY(1000 - 1);
+          });
+
+          it("we don't immediately hide the header", () => {
+            expect(header.classList).toContain("website-header");
+            expect(header.classList).toContain("website-header--revealed");
+            expect(header.classList.length).toBe(2);
+          });
+        });
+
+        describe("When we then scroll down the page by three pixels", () => {
+          beforeEach(() => {
+            simulateScrollingToY(1000);
+          });
+
+          it("we let the header return to it's natural position", () => {
+            expect(header.classList).toContain("website-header");
+            expect(header.classList.length).toBe(1);
+          });
+        });
+      });
     });
   });
 });
+
+function createMockHeader() {
+  const header = window.document.createElement("header");
+  header.classList.add("website-header");
+  window.document.body.appendChild(header);
+}
+
+function simulateScrollingToY(y) {
+  window.scrollY = y;
+
+  window.document.dispatchEvent(
+    new Event("scroll", { bubbles: true, cancelable: true })
+  );
+
+  jest.runAllTimers();
```

Comment:
We could store this in a small function describing the use, i.e triggerMockRequestAnimationFrame

Reply:
Nice one, I've done this https://github.com/codurance/site/pull/1879/commits/a7990b27df8d3eefd383867bf9eaaa33d84af18c